addons/point_of_sale/security/point_of_sale_security.xml

```diff
@@ -13,6 +13,43 @@
         <field name="users" eval="[(4, ref('base.user_root'))]"/>
     </record>
 
+    <record id="rule_pos_bank_statement" model="ir.rule">
+        <field name="name">Point Of Sale Bank Statement Rule</field>
+        <field name="model_id" ref="account.model_account_bank_statement" />
+	<field name="groups" eval="[(4, ref('group_pos_user'))]"/>
+        <field name="domain_force">[('user_id', '=', user.id)]</field>
+    </record>
+    <record id="rule_pos_bank_statement_manager" model="ir.rule">
+        <field name="name">Point Of Sale Bank Statement manager Rule</field>
+        <field name="model_id" ref="account.model_account_bank_statement" />
+	<field name="groups" eval="[(4, ref('group_pos_manager'))]"/>
+        <field name="domain_force">[(1, '=', 1)]</field>
+    </record>
+    <record id="rule_pos_bank_statement_line" model="ir.rule">
+        <field name="name">Point Of Sale Bank Statement Rule</field>
+        <field name="model_id" ref="account.model_account_bank_statement_line" />
+	<field name="groups" eval="[(4, ref('group_pos_user'))]"/>
+        <field name="domain_force">[('statement_id.user_id', '=', user.id)]</field>
+    </record>
+    <record id="rule_pos_bank_statement_line_manager" model="ir.rule">
+        <field name="name">Point Of Sale Bank Statement manager Rule</field>
+        <field name="model_id" ref="account.model_account_bank_statement_line" />
+	<field name="groups" eval="[(4, ref('group_pos_manager'))]"/>
+        <field name="domain_force">[(1, '=', 1)]</field>
+    </record>
+    <record id="rule_pos_cashbox_line" model="ir.rule">
+        <field name="name">Point Of Sale cashbox line Rule</field>
+        <field name="model_id" ref="account.model_account_cashbox_line" />
+	<field name="groups" eval="[(4, ref('group_pos_user'))]"/>
+        <field name="domain_force">[('bank_statement_id.user_id', '=', user.id)]</field>
+    </record>
+    <record id="rule_pos_cashbox_line_manager" model="ir.rule">
+        <field name="name">Point Of Sale cashbox line manager Rule</field>
+        <field name="model_id" ref="account.model_account_cashbox_line" />
+	<field name="groups" eval="[(4, ref('group_pos_manager'))]"/>
+        <field name="domain_force">[(1, '=', 1)]</field>
+    </record>
+
```

Comment:
Regarding account models: Both POS/Users and POS/Managers should have the same rights: They cannot create/edit/delete bank statements if they are not accountant (except their own).
Therefore, remove all [(1, '=', 1 )] rules.